kv/memdb/arena.go

```diff
@@ -44,13 +47,57 @@ type arena struct {
 	blocks    []arenaBlock
 }
 
+type arenaSnapshot struct {
+	blockSize     int
+	blocks        int
+	availIdx      int
```

Comment:
Seems `availIdx` is always equal to `blocks-1`?

Reply:
https://github.com/bobotu/tidb/blob/14c3dd17e6bd0c5b578eeffe3e1b3c532224f58e/kv/memdb/arena.go#L106

If `size` larger than `maxBlockSize` (128M), `avaliIdx` will less than `len(blocks) - 1`.

But will such a large value appear in practice? If not, we can remove this part of the logic to simplify the code.

Reply:
We have a 6MB entry limit, so it does not happen, we can simplify the code.